dexmaker-mockito-inline/src/main/java/com/android/dx/mockito/inline/InlineDexmakerMockMaker.java

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.dx.mockito.inline;
+
+import com.android.dx.stock.ProxyBuilder;
+import com.android.dx.stock.ProxyBuilder.MethodSetEntry;
+
+import org.mockito.exceptions.base.MockitoException;
+import org.mockito.internal.configuration.plugins.Plugins;
+import org.mockito.internal.creation.instance.Instantiator;
+import org.mockito.internal.util.Platform;
+import org.mockito.internal.util.concurrent.WeakConcurrentMap;
+import org.mockito.invocation.MockHandler;
+import org.mockito.mock.MockCreationSettings;
+import org.mockito.plugins.MockMaker;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Proxy;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Generates mock instances on Android's runtime that can mock final methods.
+ *
+ * <p>This is done by transforming the byte code of the classes to add method entry hooks.
+ */
+public final class InlineDexmakerMockMaker implements MockMaker {
+    private static final String DISPATCHER_CLASS_NAME =
+            "com.android.dx.mockito.inline.MockMethodDispatcher";
+    private static final String DISPATCHER_JAR = "dispatcher.jar";
+
+    /** {@link com.android.dx.mockito.inline.JvmtiAgent} set up during one time init */
+    private static final JvmtiAgent AGENT;
+
+    /** Error  during one time init or {@code null} if init was successful*/
+    private static final Throwable INITIALIZATION_ERROR;
+
+    /**
+     * Class injected into the bootstrap classloader. All entry hooks added to methods will call
+     * this class.
+     */
+    private static final Class DISPATCHER_CLASS;
+
+    /*
+     * One time setup to allow the system to mocking via this mock maker.
+     */
+    static {
+        // Allow to mock package private classes
+        System.setProperty("dexmaker.share_classloader", "true");
```

Comment:
Is this needed? The only reference I can find to it seems to be in Google's internal copy of dexmaker:
https://android.googlesource.com/platform/external/dexmaker/+/db20bbcc82de39f499a804d215851995041f3bcf%5E%21/#F0